### ISSUE TITLE
GitBucket gives internal server error after creating new repo with the same name as fork

### ISSUE THREAD

**Issue by bviktor** [gitbucket/gitbucket#2041]
Repro:

create repo and initialize with readme under group as

~~~
sysadmin/testrepo
~~~

then fork it under your user with the same name:

~~~
bviktor/testrepo
~~~

create a new repo under your user with the same name:

~~~
bviktor/testrepo
~~~

this tells you that

> Repository already exists.

but if you try it like this:

~~~
bviktor/testrepo.git
~~~

It will actually "succeed".

Now go to settings/danger zone and delete the repo.

This results in an internal server error as explained [here](https://github.com/gitbucket/gitbucket/issues/1976#issuecomment-384396058).

**TLDR** GitBucket only checks for name collisions literally, but passes checks when you use the `.git` suffix.

Workaround: create `bviktor/testrepo.git` again and go to your profile (now it won't result in internal server error). Then delete `testrepo`, then `testrepo.git` (in this order).

GitBucket 4.25.0
